src/req-wrap.h

```diff
@@ -23,7 +23,9 @@ class ReqWrap : public AsyncWrap {
  private:
   friend class Environment;
   ListNode<ReqWrap> req_wrap_queue_;
-  T req_;
+
+ protected:
+  T req_;  // Must be last.
```

Comment:
`req_wrap_queue_` needs to be at a fixed offset from the start of the struct because it is used by ContainerOf to calculate the address of the embedding ReqWrap.

ContainerOf compiles down to simple, fixed pointer arithmetic.  `sizeof(req_)` depends on the type of T so `req_wrap_queue_` would no longer be at a fixed offset if it came after `req_`.

It's a bit of a hack in that it depends on the layout in memory but I deemed it less worse than the untyped queue that we used before.